pyccel/codegen/printing/fcode.py

```diff
@@ -1278,6 +1278,24 @@ def _print_NumpyFull(self, expr):
         # Create statement for initialization
         init_value = self._print(expr.fill_value)
         return init_value
+    
+    def _print_NumpyAmax(self, expr):
+        # Assuming expr.arg is an array
+        array_arg = expr.arg
+        if array_arg.dtype is NativeBool():
+            arg_code = self._print(PythonInt(array_arg))
+        else:
+            arg_code = self._print(array_arg)
+
+        if array_arg.dtype is NativeComplex():
+            test_arg = PythonReal(array_arg)
+            idx_variable = self.scope.get_temporary_variable(NativeInteger(), rank = 1)
```

Comment:
Looks like I missed a detail:
```suggestion
            idx_variable = self.scope.get_temporary_variable(NativeInteger(), rank = 1, class_type = NumpyNDArrayType())
```
This probably also needs importing (from ast.numpyext)